src/main/java/fr/zcraft/quartzlib/components/i18n/translators/gettext/PluralForms.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Plugin UHCReloaded : Alliances
+ *
+ * Copyright ou © ou Copr. Amaury Carrade (2016)
+ * Idées et réflexions : Alexandre Prokopowicz, Amaury Carrade, "Vayan".
+ *
+ * Ce logiciel est régi par la licence CeCILL soumise au droit français et
+ * respectant les principes de diffusion des logiciels libres. Vous pouvez
+ * utiliser, modifier et/ou redistribuer ce programme sous les conditions
+ * de la licence CeCILL telle que diffusée par le CEA, le CNRS et l'INRIA
+ * sur le site "http://www.cecill.info".
+ *
+ * En contrepartie de l'accessibilité au code source et des droits de copie,
+ * de modification et de redistribution accordés par cette licence, il n'est
+ * offert aux utilisateurs qu'une garantie limitée.  Pour les mêmes raisons,
+ * seule une responsabilité restreinte pèse sur l'auteur du programme,  le
+ * titulaire des droits patrimoniaux et les concédants successifs.
+ *
+ * A cet égard  l'attention de l'utilisateur est attirée sur les risques
+ * associés au chargement,  à l'utilisation,  à la modification et/ou au
+ * développement et à la reproduction du logiciel par l'utilisateur étant
+ * donné sa spécificité de logiciel libre, qui peut le rendre complexe à
+ * manipuler et qui le réserve donc à des développeurs et des professionnels
+ * avertis possédant  des  connaissances  informatiques approfondies.  Les
+ * utilisateurs sont donc invités à charger  et  tester  l'adéquation  du
+ * logiciel à leurs besoins dans des conditions permettant d'assurer la
+ * sécurité de leurs systèmes et ou de leurs données et, plus généralement,
+ * à l'utiliser et l'exploiter dans les mêmes conditions de sécurité.
+ *
+ * Le fait que vous puissiez accéder à cet en-tête signifie que vous avez
+ * pris connaissance de la licence CeCILL, et que vous en avez accepté les
+ * termes.
+ */
+
+package fr.zcraft.quartzlib.components.i18n.translators.gettext;
+
+import fr.zcraft.quartzlib.tools.PluginLogger;
+import java.util.Locale;
+import java.util.function.Function;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class PluralForms {
+    /**
+     * The default plural forms function when we cannot do anything else. It's the English plural rule.
+     */
+    private static final Function<Long, Integer> FORMS_FUNCTION_FALLBACK = n -> n != 1 ? 1 : 0;
+
+    /**
+     * The number of plural forms for this script.
+     */
+    private final int formsCount;
+
+    /**
+     * The form script, as written in the PO/MO file.
+     */
+    @NotNull
+    private final String formsScript;
+
+    /**
+     * A function to call to compute the correct plural form
+     * from a count.
+     */
+    private final Function<Long, Integer> formsFunction;
```

Comment:
Why form is at plural  here ? In the documentation it's at singular

Reply:
It's the function that handle plural formS. It compute the correct form to use among all formS available. Hence the -s but I'm not that attached to it.